### PR TITLE
graalvm-ce

### LINKED ISSUE #81: lmages with GraalVM CE wanted
https://hub.docker.com/r/oracle/graalvm-ce

### PR DESCRIPTION
**Pull request by francisdb**
fixes #81

### CODE REVIEW

**Comment by francisdb** [comment]
needs an update once #90 is accepted

**Comment by francisdb** [comment]
@hseeberger any chance this pr will make it or should I just close it (and the related ticket)?

**Comment by gurinderu** [comment]
19.3.0 already available

File: hooks/vars.sh [hunk 1]
```diff
@@ -7,4 +7,6 @@ TAG="11.0.4_1.3.3_2.13.1" DOCKER_CONTEXT="debian" BASE_IMAGE_TAG="11.0.4-jdk-str
 TAG="11.0.4_1.3.3_2.12.10" DOCKER_CONTEXT="debian" BASE_IMAGE_TAG="11.0.4-jdk-stretch" SBT_VERSION="1.3.3" SCALA_VERSION="2.12.10"
 TAG="11.0.2-oraclelinux7_1.3.3_2.13.1" DOCKER_CONTEXT="oracle" BASE_IMAGE_TAG="11.0.2-jdk-oraclelinux7" SBT_VERSION="1.3.3" SCALA_VERSION="2.13.1"
 TAG="11.0.2-oraclelinux7_1.3.3_2.12.10" DOCKER_CONTEXT="oracle" BASE_IMAGE_TAG="11.0.2-jdk-oraclelinux7" SBT_VERSION="1.3.3" SCALA_VERSION="2.12.10"
+TAG="graalvm-ce19.3.0_1.3.3_2.13.1" DOCKER_CONTEXT="graalvm-ce" BASE_IMAGE_TAG="19.3.0" SBT_VERSION="1.3.3" SCALA_VERSION="2.13.1"
```
Comment (hseeberger) [inline hooks/vars.sh]: s/2.13.0/2.13.1

File: hooks/vars.sh [hunk 2]
```diff
@@ -7,4 +7,6 @@ TAG="11.0.4_1.3.3_2.13.1" DOCKER_CONTEXT="debian" BASE_IMAGE_TAG="11.0.4-jdk-str
...
+TAG="graalvm-ce19.3.0-java11_1.3.3_2.13.1" DOCKER_CONTEXT="graalvm-ce" BASE_IMAGE_TAG="19.3.0-java11" SBT_VERSION="1.3.3" SCALA_VERSION="2.13.1"
```
Comment (hseeberger) [inline hooks/vars.sh]: s/2.13.0/2.13.1

Comment (francisdb) [reply hooks/vars.sh]: you're too fast :-)

**Comment by gurinderu** [comment]
@francisdb 
I think it would be better if you add explicit tags for java 8 and java 11
i.e
```
TAG="graalvm-ce-19.3.0-java11_1.3.3_2.13.1" DOCKER_CONTEXT="graalvm-ce" BASE_IMAGE_TAG="19.3.0-java11" SBT_VERSION="1.3.3" SCALA_VERSION="2.13.1"
TAG="graalvm-ce-19.3.0-java11_1.3.3_2.12.10" DOCKER_CONTEXT="graalvm-ce" BASE_IMAGE_TAG="19.3.0-java11" SBT_VERSION="1.3.3" SCALA_VERSION="2.12.10"
TAG="graalvm-ce-19.3.0-java8_1.3.3_2.13.1" DOCKER_CONTEXT="graalvm-ce" BASE_IMAGE_TAG="19.3.0-java8" SBT_VERSION="1.3.3" SCALA_VERSION="2.13.1"
TAG="graalvm-ce-19.3.0-java8_1.3.3_2.12.10" DOCKER_CONTEXT="graalvm-ce" BASE_IMAGE_TAG="19.3.0-java8" SBT_VERSION="1.3.3" SCALA_VERSION="2.12.10"

```

**Comment by francisdb** [comment]
@gurinderu true, let me fix that

**Comment by hseeberger** [comment]
+1 for tags explicit wrt Java version.

**Comment by francisdb** [comment]
@hseeberger is there a way to validate that the build on dockerhub will work?

**Comment by francisdb** [comment]
do we really need those 2.12.10 for graalvm-ce?

**Comment by hseeberger** [comment]
> @hseeberger is there a way to validate that the build on dockerhub will work?

No idea :-(

**Comment by hseeberger** [comment]
> do we really need those scala 2.12.10 versions for graalvm-ce?

Why not?

**Comment by francisdb** [comment]
ok, one more update on it's way

**Comment by francisdb** [comment]
what do you think about adding

```
TAG="latest" DOCKER_CONTEXT="debian" BASE_IMAGE_TAG="11.0.4-jdk-stretch" SBT_VERSION="1.3.3" SCALA_VERSION="2.13.1"
```

File: graalvm-ce/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  echo "case object Temp" > Temp.scala && \
+  sbt compile && \
+  rm -r project && rm build.sbt && rm Temp.scala && rm -r target
+
+# Link everything into root as well
+# This allows users of this container to choose, whether they want to run the container as sbtuser (non-root) or as root
+USER root
+RUN \
+  echo "export PATH=/home/sbtuser/scala-$SCALA_VERSION/bin:$PATH" >> /root/.bashrc && \
+  ln -s /home/sbtuser/.ivy2 /root/.ivy2 && \
```
Comment (gurinderu) [inline graalvm-ce/Dockerfile]: Looks like another dockerfiles has this line 
ln -s /home/sbtuser/.cache /root/.cache && \
but I'm not sure that it is needed

Comment (francisdb) [reply graalvm-ce/Dockerfile]: good point, going to sync with the oracle image

**Comment by francisdb** [comment]
I can do a last rebase when you think everything is ok

**Comment by hseeberger** [comment]
> what do you think about adding
> 
> ```
> TAG="latest" DOCKER_CONTEXT="debian" BASE_IMAGE_TAG="11.0.4-jdk-stretch" SBT_VERSION="1.3.3" SCALA_VERSION="2.13.1"
> ```

I don't see why exactly this should be "latest".

File: hooks/vars.sh [hunk 4]
```diff
@@ -1,10 +1,15 @@
 #!/bin/bash
 
 export VARS='
+TAG="latest" DOCKER_CONTEXT="debian" BASE_IMAGE_TAG="11.0.4-jdk-stretch" SBT_VERSION="1.3.3" SCALA_VERSION="2.13.1"
```
Comment (hseeberger) [inline hooks/vars.sh]: Please remove.

Comment (francisdb) [reply hooks/vars.sh]: was mainly to also tackle  #93 but if you don't want that you should probably just close that ticket

**Comment by jtjeferreira** [comment]
Hi @hseeberger ! Thanks for the nice images. However for some reason I cant see any `java11` builds in https://hub.docker.com/r/hseeberger/scala-sbt/tags...

**Pull request merged**